lib/connection.js

```diff
@@ -70,7 +70,12 @@ function Connection(base) {
   this._closeCalled = false;
   this._hasOpened = false;
   this.plugins = [];
-  this.id = id++;
+  // this.id = id++;
+  if (typeof base === 'undefined' || !base.connections.length) {
```

Comment:
I'd just get rid of this `if` statement, shouldn't be necessary because `Connection` constructor is private so `base` should always be a Mongoose instance.

Reply:
![capture](https://user-images.githubusercontent.com/39510674/112496775-63c00100-8d5b-11eb-96ed-9460f563edcf.png)
without this if statement these errors pop up